grails-app/views/inventoryLevel/_form.gsp

```diff
@@ -32,10 +32,11 @@
                     <label for="inventory.id"><warehouse:message code="inventoryLevel.inventory.label" default="Inventory" /></label>
                 </td>
                 <td valign="top" class="value ${hasErrors(bean: inventoryLevelInstance, field: 'inventory', 'errors')}">
-                    <g:select id="${inventoryLevelInstance?.id?'edit':'save'}-${inventoryLevelInstance?.id}-inventory-id" name="inventory.id" from="${org.pih.warehouse.inventory.Inventory.list()}" optionKey="id"
+                    <g:set var="location" value="${Inventory.list().find { it.warehouse == currentLocation } }"/>
+                    <g:select id="${inventoryLevelInstance?.id?'edit':'save'}-${inventoryLevelInstance?.id}-inventory-id" name="inventory.id" from="${Inventory.list()}" optionKey="id"
                                 style="width: 300px;"
-                                noSelection="['null':warehouse.message(code:'inventoryLevel.chooseInventory.label',default:'Choose inventory')]"
-                                value="${inventoryLevelInstance?.inventory?.id}" class="chzn-select-deselect" />
+                                noSelection="['':'']"
+                                value="${inventoryLevelInstance?.inventory?.id?:location.id}" class="chzn-select-deselect" />
```

Comment:
This needs to be an Inventory object, not a Location (so `location?.inventory?.id`). However, you'll get a lazy initialization exception if you reference the inventory association here without rehydrating the location object, so you'll need to do something like this:
```
render(view: "create", model: [inventoryLevelInstance: inventoryLevelInstance, currentLocation: Location.get(session.warehouse.id)]
```
Also, make sure you're handling all use cases i.e. Edit / Create, as well as when either of those use cases has validation errors from both Product > Inventory Levels and Edit Product > Stock Levels tab).

Reply:
Ah, I see that the object is an Inventory object. However this is still the wrong approach. See comment above where you use `<g:set/>` to assign the inventory object to the `location` variable.